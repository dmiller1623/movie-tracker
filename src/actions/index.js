export const populateMovies = (movies) => ({
  type: 'POPULATE_MOVIES',
  movies
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
});

export const signUpUser = (email, password) => ({
  type: 'SIGNUP_USER',
  email,
  password
});

export const signOutUser = () => {
  type: 'SIGN_OUT'

}
