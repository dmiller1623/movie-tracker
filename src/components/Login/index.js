import React, { Component } from 'react';

export default class Login extends Component {
  constructor(){
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value});
  }

  render() {
    return (
      <div>
        <form>
          <input
            name='email'
            value={this.state.email}
            placeholder='email'
            onChange={(event) => this.handleChange(event)}
          />
          <input
            name='password'
            value={this.state.password}
            placeholder='Password'
            onChange={(event) => this.handleChange(event)}
          />
          <button>Submit</button>
        </form>
      </div>
    );
  }
}