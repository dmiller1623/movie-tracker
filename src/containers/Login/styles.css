form {
  display: flex;
  flex-direction: column;
  width: 35%;
  margin: 8rem auto 0 auto; }

input {
  height: 2.5rem;
  font-size: 1.8rem;
  padding-left: .5rem;
  font-family: "Orbitron", sans-serif;
  margin-bottom: 1rem; }
  input::placeholder {
    color: #93A2AA; }

button {
  outline: none;
  height: 2.5rem;
  margin-top: 2rem;
  font-family: 'Orbitron', sans-serif;
  font-size: 1.8rem;
  padding: .5rem;
  background-color: #93A2AA;
  color: #f2f2f2;
  border: none;
  border-radius: 10px;
  cursor: pointer; }
  button:hover {
    background-color: #6c797f; }

@media only screen and (max-width: 1000px) {
  form {
    width: 75%; } }

@media only screen and (max-width: 450px) {
  form {
    width: 90%; } }
