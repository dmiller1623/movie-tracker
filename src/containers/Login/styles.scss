@import "variables.scss";

form {
  display: flex;
  flex-direction: column;
  width: 35%;
  margin: 8rem auto 0 auto;
}

input {
  height: 2.5rem;
  font-size: 1.8rem;
  padding-left: .5rem;
  font-family: $main-font;
  margin-bottom: 1rem;
  &::placeholder {
    color: $light-grey;
  }
}

button {
  outline: none;
  height: 2.5rem;
  margin-top: 2rem;
  font-family: 'Orbitron', sans-serif;
  font-size: 1.8rem;
  padding: .5rem;
  background-color: $light-grey;
  color: $off-white;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  &:hover {
    background-color: #6c797f;
  }
}

@media only screen and (max-width: 1000px) { 
  form {
    width: 75%;
  }
}

@media only screen and (max-width: 450px) { 
  form {
    width: 90%;
  }
}
