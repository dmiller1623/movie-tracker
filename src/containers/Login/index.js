import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';
import { getUser } from '../../utilities/apiCalls/apiCalls';


class Login extends Component {
  constructor(){
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value});
  }

  submitLogin = async (event) => {
    event.preventDefault();
    const { email, password } = this.state;
    if(email && password) {
    const cleanEmail = email.trim().toLowerCase()
    const user = await getUser(cleanEmail, password)
    this.props.loginUser(user);
    } else {
      alert('Please enter both email and password')
    }
  }

  render() {
    return (
      <div>
        <form onSubmit={(event) => this.submitLogin(event)}>
          <input
            type="email"
            name='email'
            value={this.state.email}
            placeholder='email address'
            onChange={(event) => this.handleChange(event)}
          />
          <input
            name='password'
            value={this.state.password}
            placeholder='Password'
            onChange={(event) => this.handleChange(event)}
          />
          <button>Submit</button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  loginUser: (email, password) => dispatch(loginUser(email, password))
});

export default connect(null, mapDispatchToProps)(Login)