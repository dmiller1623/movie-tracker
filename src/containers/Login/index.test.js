import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapDispatchToProps } from '../Login';
import { loginUser, populateFavorites } from '../../actions';


describe('Login', () => {
  let wrapper;
  const mockUser = {id: 1, name: "Taylor", password: "password", email: "tman2272@aol.com"}
  describe('Login component', () => {

  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function loginUser', () => {

      const mockDispatch = jest.fn();
      const actionToDispatch = loginUser(mockUser);

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.loginUser(mockUser);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

    });
    it('should call dispatch when using a function populateFavorites', () => {
      const mockFavorites = [0, 1, 2];

      const mockDispatch = jest.fn();
      const actionToDispatch = populateFavorites(mockFavorites);

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.populateFavorites(mockFavorites);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

    });
  })
})