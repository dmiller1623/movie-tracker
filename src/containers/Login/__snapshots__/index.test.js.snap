// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <input
                autoComplete="on"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
        />
        <input
                autoComplete="off"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
        />
        <button>
                Login
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            autoComplete="on"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
/>,
          <input
            autoComplete="off"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button>
            Login
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "on",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <input
                    autoComplete="on"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
          />
          <input
                    autoComplete="off"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <button>
                    Login
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoComplete="on"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
/>,
            <input
              autoComplete="off"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <button>
              Login
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "on",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
